### PR TITLE
klee 3.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.
<details>
  <summary>release notes</summary>
  <pre>KLEE 3.1, 29 February 2024
==========================

Incorporating changes from 8 June 2023 to 29 February 2024.
Maintainers during this time span: @ccadar, @MartinNowack, @251.
Documentation at http://klee.github.io/releases/docs/v3.1

## Major features and important changes
- New execution tree implementation and klee-exec-tree tool (@251)
- KDAlloc is now the default allocator in KLEE (KDAlloc was introduced in KLEE 3.0)
- Resolve memory reads/writes to single objects in more cases (@tkuchta)
- Concretize values based on seeds when available (@ccadar)
- Fixed some interactions with external code (@ccadar, @MartinNowack, mishok2503)

## LLVM support
- Current recommended version is still LLVM 13
- Added support for LLVM 15 and 16 (@MartinNowack)
- Removed support for LLVM <11 (@danielschemmel)
- KLEE 3.1 will be the last version with support for LLVM <13

## Options, scripts and KLEE intrinsics added, changed or removed
- New klee-exec-tree tool (@251)
- New --write-exec-tree and --exec-tree-batch-size options (@251)
- Renamed --compress-process-tree to --compress-exec-tree (@ccadar)
- PTree is now called ExecutionTree in the code (@ccadar)
- KDAlloc (--kdalloc) is now enabled by default (@ccadar)
- Replaced --suppress-external-warnings and --all-external-warnings with --external-call-warnings=none|once-per-function|all (@ccadar)
- Keep in the KLEE and Kleaver help menus only the KLEE/Kleaver option categories (@ccadar)
- Removed --cex-cache-exp, a broken experimental optimisation for validity (@ccadar)
- Removed --zero-seed-extension, and merge it with --allow-seed-extension (@ccadar)

## Other changes
- Improvements to KDAlloc (@danielschemmel)
- Avoid generating array names in solver builders that could accidently collide (@MartinNowack)
- Function has_permission in the POSIX model now returns permission error a single time in symbolic execution mode (@ccadar)
- Smaller refactorings, fixes and improvements, test cases, maintenance, comments, web version, website, etc. (@251, @ccadar, @dependabot, @danielschemmel, @davidtr1037, @jbuening, @marco6, @MartinNowack, @McSinyx, @sp1ff)
</pre>
</details>

### CODE REVIEW

**Comment by chenrui333** [comment]
- https://github.com/klee/klee/pull/1664

**Comment by chenrui333** [comment]
still build issue

```
-- klee-libc++ support enabled
CMake Error at CMakeLists.txt:510 (find_file):
  Could not find KLEE_LIBCXX_BC_PATH using the following files: libc++.bca,
  libc++.so.bc, libc++.dylib.bc
```

File: Formula/k/klee.rb [hunk 1]
```diff
@@ -105,13 +104,15 @@ def install
     # https://github.com/klee/klee/blob/v#{version}/README-CMake.md
     args = %W[
       -DKLEE_RUNTIME_BUILD_TYPE=Release
+      -DENABLE_KLEE_LIBCXX=ON
       -DKLEE_LIBCXX_DIR=#{libcxx_install_dir}
-      -DKLEE_LIBCXX_INCLUDE_DIR=#{libcxx_install_dir}/include/c++/v1
+      -DKLEE_LIBCXX_BC_PATH=#{libcxx_install_dir}/lib
+      -DKLEE_LIBCXX_INCLUDE_DIR=#{libcxx_install_dir}/include
+      -DKLEE_LIBCXX_INCLUDE_PATH=#{libcxx_install_dir}/include/c++/v1
```
Comment (cho-m) [inline Formula/k/klee.rb:111]: It looks like upstream removed some cache variables in https://github.com/klee/klee/commit/f27b748ba2be259177aa672e4d5fc39d36479c7f#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL509

However, based on documentation, it doesn't seem that was intended. So may need to report upstream.

As workaround, `find_file`/`find_path` creates cache entries so I added the `KLEE_LIBCXX_BC_PATH` and `KLEE_LIBCXX_INCLUDE_PATH` to bypass this issue.

**Comment by cho-m** [comment]
Going to revert back to `llvm@14`. Currently, it looks like `llvm@16` gets stuck when running `klee` on Intel runners. Can retry LLVM update in follow up PR.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**